dashboard/test/models/concerns/user/verifiable_test.rb

```diff
@@ -0,0 +1,83 @@
+require 'test_helper'
+
+class VerifiableTest < ActiveSupport::TestCase
+  include Minitest::RSpecMocks
+
+  describe '#verify_teacher!' do
+    let(:teacher) {create :teacher}
+
+    before do
+      UserPermission.any_instance.stubs(:send_verified_teacher_email)
+      teacher.verify_teacher!
+    end
+
+    it 'sets the permission correctly' do
+      permission_values = teacher.permissions.map(&:permission)
+      _(permission_values).must_include UserPermission::AUTHORIZED_TEACHER
+    end
+  end
+
+  describe '#verified_teacher?' do
+    subject(:verified_teacher?) {user.verified_teacher?}
+    context 'when the user is a normal teacher' do
+      let(:user) {create :teacher}
```

Comment:
In your other PR, we discussed the use of explicit over implicit factory calls.
I will create a task for updating these test later on, but for now I'm just flaging it.